x-pack/plugins/security_solution/public/management/pages/event_filters/view/components/flyout/index.tsx

```diff
@@ -33,97 +32,125 @@ import {
 import { getInitialExceptionFromEvent } from '../../../store/utils';
 
 export interface EventFiltersFlyoutProps {
-  data?: Ecs;
+  type?: 'create' | 'edit';
+  id?: string;
   onCancel(): void;
 }
 
-export const EventFiltersFlyout: React.FC<EventFiltersFlyoutProps> = memo(({ data, onCancel }) => {
-  useEventFiltersNotification();
-  const dispatch = useDispatch<Dispatch<AppAction>>();
-  const formHasError = useEventFiltersSelector(getFormHasError);
-  const creationInProgress = useEventFiltersSelector(isCreationInProgress);
-  const creationSuccessful = useEventFiltersSelector(isCreationSuccessful);
+export const EventFiltersFlyout: React.FC<EventFiltersFlyoutProps> = memo(
+  ({ onCancel, id, type = 'create' }) => {
+    useEventFiltersNotification();
+    const dispatch = useDispatch<Dispatch<AppAction>>();
+    const formHasError = useEventFiltersSelector(getFormHasError);
+    const creationInProgress = useEventFiltersSelector(isCreationInProgress);
+    const creationSuccessful = useEventFiltersSelector(isCreationSuccessful);
 
-  useEffect(() => {
-    if (creationSuccessful) {
-      onCancel();
-      dispatch({
-        type: 'eventFiltersFormStateChanged',
-        payload: {
-          type: 'UninitialisedResourceState',
-        },
-      });
-    }
-  }, [creationSuccessful, onCancel, dispatch]);
-
-  // Initialize the store with the event passed as prop to allow render the form. It acts as componentDidMount
-  useEffect(() => {
-    dispatch({
-      type: 'eventFiltersInitForm',
-      payload: { entry: getInitialExceptionFromEvent(data) },
-    });
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+    useEffect(() => {
+      if (creationSuccessful) {
+        onCancel();
+        dispatch({
+          type: 'eventFiltersFormStateChanged',
+          payload: {
+            type: 'UninitialisedResourceState',
+          },
+        });
+      }
+    }, [creationSuccessful, onCancel, dispatch]);
 
-  const handleOnCancel = useCallback(() => {
-    if (creationInProgress) return;
-    onCancel();
-  }, [creationInProgress, onCancel]);
+    // Initialize the store with the id passed as prop to allow render the form. It acts as componentDidMount
+    useEffect(() => {
+      if (type === 'edit' && !!id) {
+        dispatch({
+          type: 'eventFiltersInitFromId',
+          payload: { id },
+        });
+      } else {
```

Comment:
What if type is `edit`, but `id` is not defined or empty string?
If I read this correctly, this `else` branch would automatically assume we are in "create" mode although the url param might be `edit`?

Reply:
There is a useEffect that cleans the url query params when they are wrong at event_filters_list_page.tsx